### PR TITLE
[Php81] Allow normal variable in trait on NullToStrictStringFuncCallArgRector

### PR DESCRIPTION
**Pull request by samsonasik**
Continue on https://github.com/rectorphp/rector-src/pull/3180

The following normal variable:

```php
trait NormalVariableInTrait
{
    public function getTitle($title)
    {
        return trim($title);
    }
}
```

should be processed to :

```diff
-return trim($title);
+return trim((string) $title);
```

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Good one :+1:  Go for the merge :heavy_check_mark:

**Comment by samsonasik** [comment]
I rolled back property assigned in trait construct that was processed on https://github.com/rectorphp/rector-src/pull/3173 to be skipped back, while `__construct()` method can be overridden by class that consume the trait, the use case seems very rare, so skip seems make sense:

  https://github.com/rectorphp/rector-src/pull/3181/commits/241204c92f7dfe2622d0a070f2591b5edddfd9e2

File: rules/Php81/Rector/FuncCall/NullToStrictStringFuncCallArgRector.php [hunk 1]
```diff
@@ -488,6 +490,15 @@ private function processNullToStrictStringOnNodePosition(
         return $funcCall;
     }
 
+    private function shouldSkipTrait(Expr $expr, bool $isTrait): bool
+    {
+        if (! $expr instanceof MethodCall) {
```
Comment (samsonasik) [inline rules/Php81/Rector/FuncCall/NullToStrictStringFuncCallArgRector.php:495]: The `MethodCall` or possibly `StaticCall` detection as FuncCall argument may be in next iteration for improvement where the result is Typed and actually non-mixed type, eg: Nullable type.

**Comment by samsonasik** [comment]
I am merging it ;)

**Pull request merged**